cmd/snap/cmd_run.go

```diff
@@ -954,3 +967,88 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		return syscallExec(cmd[0], cmd, env)
 	}
 }
+
+func randomUUID() (string, error) {
+	uuidBytes, err := ioutil.ReadFile("/proc/sys/kernel/random/uuid")
+	return strings.TrimSpace(string(uuidBytes)), err
+}
+
+var createTransientScope = func(securityTag string) error {
+	if !features.RefreshAppAwareness.IsEnabled() {
+		return nil
+	}
+
+	// The scope is created with a DBus call to systemd running either on
+	// system or session bus, depending on if we are starting a program as root
+	// or as a regular user.
+	var conn *dbus.Conn
+	var err error
+	if os.Getuid() == 0 {
+		conn, err = dbus.SystemBus()
+	} else {
+		conn, err = dbus.SessionBus()
+	}
+	if err != nil {
+		return err
+	}
+
+	// The property and auxUnit types are not documented but can be traced from
```

Comment:
It's curious, this changed since I looked at it first.
```
StartTransientUnit() may be used to create and start a transient unit, which will be released as soon as it is not running or referenced anymore or the system is rebooted. name is the unit name including suffix, and must be unique. mode is the same as in StartUnit(), properties contains properties of the unit, specified like in SetUnitProperties(). aux is currently unused and should be passed as empty array. See the New Control Group Interfaces for more information how to make use of this functionality for resource control purposes.
```

That said, I'm almost sure `aux` is false, having read the code for the implementation I didn't see `aux` being ignored. In either case I will link to the documentation and double check.